### PR TITLE
fix(ras-acc): add onSuccess/onError callbacks to recaptcha render method

### PR DESCRIPTION
**Pull request by chickenn00dle**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Part of https://github.com/Automattic/newspack-blocks/pull/1912.

### How to test the changes in this Pull Request:

See https://github.com/Automattic/newspack-blocks/pull/1912 for instructions

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/other-scripts/recaptcha/index.js [hunk 1]
```diff
@@ -135,53 +136,39 @@ function renderWidget( form, errorHandler = null ) {
...
 		setInterval( () => refreshWidget( button ), 120000 ); // Refresh widget every 2 minutes.
 
-		// Refresh reCAPTCHAs on Woo checkout update and error.
-		( function ( $ ) {
-			if ( ! $ ) {
-				return;
-			}
-			$( document ).on( 'updated_checkout', () => renderWidget( form ) );
-			$( document.body ).on( 'checkout_error', () => renderWidget( form ) );
-		} )( jQuery );
```
Comment (miguelpeixe) [inline src/other-scripts/recaptcha/index.js:171]: Won't this break the reCaptcha integration with legacy checkout? My understanding is that our reCaptcha feature is not exclusive to the modal experience.

https://github.com/Automattic/newspack-plugin/blob/f9d4c5389ff6d1824844e64f91a70932efe0ecca/includes/class-recaptcha.php#L29

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
Great changes! Small question below

File: src/other-scripts/recaptcha/index.js (hunk 1 above)
Comment (chickenn00dle) [reply src/other-scripts/recaptcha/index.js:171]: It looks like this only [triggers for v3](https://github.com/Automattic/newspack-plugin/blob/f9d4c5389ff6d1824844e64f91a70932efe0ecca/includes/class-recaptcha.php#L430) where the strategy is to [add a hidden recaptcha field](https://github.com/Automattic/newspack-plugin/blob/f9d4c5389ff6d1824844e64f91a70932efe0ecca/src/other-scripts/recaptcha/index.js#L60). The `renderWidget` method is only called for v2 so changes here shouldn't have an impact.

Comment (miguelpeixe) [reply src/other-scripts/recaptcha/index.js:171]: Thank you for explaining!

File: src/other-scripts/recaptcha/index.js [hunk 2]
```diff
@@ -147,8 +152,16 @@ function renderWidget( form, onSuccess = null, onError = null ) {
...
+				const retryCount = parseInt( button.getAttribute( 'data-recaptcha-retry-count' ) ) || 0;
+				if ( retryCount < 3 ) {
+					button.setAttribute( 'data-recaptcha-retry-count', retryCount + 1 );
+					refreshWidget( button );
+				} else {
+					clearInterval( refreshIntervalId );
+				}
+				const message = retryCount < 3
+					? wp.i18n.__( 'There was an error with reCAPTCHA. Please try again.', 'newspack-plugin' )
+					: wp.i18n.__( 'There was an error with reCAPTCHA. Please reload the page and try again.', 'newspack-plugin' );
```
Comment (chickenn00dle) [inline src/other-scripts/recaptcha/index.js:164]: I wonder if we should instead nudge the reader to contact the site admin instead. I have seen cases where the API fails for other unknown reasons though (possibly due to some dev site strangeness), so am just asking for a reload.

Let me know if the site admin messaging would be better though!

Comment (miguelpeixe) [reply src/other-scripts/recaptcha/index.js:164]: This should be fine. This will happen on any page load so the admins would become aware instantly.

**Review by miguelpeixe — APPROVED** [review]

**Pull request merged**